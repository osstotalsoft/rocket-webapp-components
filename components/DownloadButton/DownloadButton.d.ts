import { ButtonProps } from '@material-ui/core/Button';

export interface DownloadButtonProps extends Omit<ButtonProps, 'color'> {
    /**
    * The title of button.
    */
    title?: string;
    /**
    * Callback fired when a "click" event is detected.
    */
    onClick?: () => void
    /**
    * If true, the button will be disabled.
    */
    disabled?: boolean
    /**
    * @default "theme"
    * The color of button.
    */
    color?: "primary" |
    "info" |
    "theme" |
    "themeNoBackground" |
    "themeWithBackground" |
    "success" |
    "warning" |
    "danger" |
    "rose" |
    "white" |
    "simple" |
    "defaultNoBackground" |
    "primaryNoBackground" |
    "infoNoBackground" |
    "successNoBackground" |
    "warningNoBackground" |
    "dangerNoBackground" |
    "roseNoBackground"
    /**
    * @default "medium"
    * The size of button.
    */
    size?: "medium" | "small"
    /**
    * @default "medium"
    * The fontSize of button's content.
    */
    fontSize?: "inherit" | "small" | "medium" | "large"
    /**
    * Override or extend the styles applied to the component
    */
    customClass?: string
    /**
    * @default "false"
    * If true, the keyboard focus ripple will be disabled. 
    */
    disableFocusRipple?: boolean
    /**
     * If true, the ripple effect will be disabled.
     */
    disableRipple?: boolean
    /**
     * @default "false"
     * If given, uses a negative margin to counteract the padding on one side 
     * (this is often helpful for aligning the left or right side of the icon 
     * with content above or below, without ruining the border size and shape).
     */
    edge?: "start" | "end" | "false"
}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/download-button
 *
 *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/DownloadButton/DownloadButton.js
 * 
 * Bit component used:
 * 
 * - IconButton: https://bit.dev/totalsoft_oss/react-mui/icon-button
 * 
 */
export default function DownloadButton(props: DownloadButtonProps): JSX.Element;