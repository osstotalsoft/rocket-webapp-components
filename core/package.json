{
  "engines": {
    "node": ">=10"
  },
  "scripts": {
    "prebuild": "rimraf build",
    "build": "npm run build:cjs &&  npm run build:copy-files",
    "build:cjs": "babel src -d build --copy-files",
    "build:copy-files": "node ./scripts/copy-files.js",
    "bit-build": "bit -v && bit import && bit build"
  },
  "devDependencies": {
    "@babel/cli": "^7.8.0",
    "@babel/core": "^7.7.7",
    "@babel/node": "^7.7.7",
    "@babel/plugin-proposal-decorators": "^7.8.3",
    "@babel/preset-env": "^7.7.7",
    "@babel/preset-react": "^7.8.0",
    "babel-eslint": "^10.0.3",
    "cross-env": "^7.0.2",
    "eslint": "^6.8.0",
    "eslint-config-prettier": "^6.7.0",
    "eslint-config-standard": "^14.1.0",
    "eslint-config-standard-react": "^9.2.0",
    "eslint-plugin-import": "^2.18.2",
    "eslint-plugin-node": "^11.0.0",
    "eslint-plugin-prettier": "^3.1.1",
    "eslint-plugin-promise": "^4.2.1",
    "eslint-plugin-react": "^7.17.0",
    "eslint-plugin-standard": "^4.0.1",
    "gh-pages": "^2.2.0",
    "npm-run-all": "^4.1.5",
    "prettier": "^1.19.1",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "^3.4.1",
    "rimraf": "^3.0.0",
    "url-search-params-polyfill": "^7.0.1"
  },
  "peerDependencies": {
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-flags": "^0.1.18",
    "react-router-dom": "^5.1.2"
  },
  "sideEffects": false,
  "dependencies": {},
  "bit": {
    "env": {
      "compiler": "totalsoft_oss.react-mui/compilers/react@1.0.32"
    },
    "componentsDefaultDirectory": "components/{name}",
    "packageManager": "npm"
  }
}
