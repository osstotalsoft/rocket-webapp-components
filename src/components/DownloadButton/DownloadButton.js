import React from "react";
import PropTypes from "prop-types";
import CloudDownloadIcon from '@material-ui/icons/CloudDownload';
import CustomIconButton from '../IconButton';

const DownloadButton = ({ title, onClick, disabled, fontSize, ...rest }) => (
    <CustomIconButton aria-label="Download" onClick={onClick} disabled={disabled} tooltip={title} {...rest}>
        <CloudDownloadIcon fontSize={fontSize} />
    </CustomIconButton>
)

DownloadButton.defaultProps = {
    color: "theme",
    fontSize: "medium",
    size: "medium"
};

DownloadButton.propTypes = {
    /**
    * The title of button.
    */
    title: PropTypes.string,
    /**
    * Callback fired when a "click" event is detected.
    */
    onClick: PropTypes.func,
    /**
    * The color of button.
    */
    color: PropTypes.oneOf([
        "primary",
        "info",
        "theme",
        "themeNoBackground",
        "themeWithBackground",
        "success",
        "warning",
        "danger",
        "rose",
        "white",
        "simple",
        "defaultNoBackground",
        "primaryNoBackground",
        "infoNoBackground",
        "successNoBackground",
        "warningNoBackground",
        "dangerNoBackground",
        "roseNoBackground",
        "themeWithGradient"
    ]),
    /**
    * The size of button.
    */
    size: PropTypes.oneOf([
        "small",
        "medium"
    ]),
    /**
    * The fontSize of button's content.
    */
    fontSize: PropTypes.oneOf([
        "inherit",
        "small",
        "medium",
        "large"
    ]),
    /**
    * If true, the button will be disabled.
    */
    disabled: PropTypes.bool
};

export default DownloadButton;