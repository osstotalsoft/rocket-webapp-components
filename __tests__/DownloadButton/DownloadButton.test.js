import React from 'react';
import DownloadButton from '../../components/DownloadButton/DownloadButton';
import { mount } from 'enzyme';
import { IconButton } from "@material-ui/core";

describe('DownloadButton', () => {
    it('onClick should be called', () => {
        const onClickMock = jest.fn();

        const wrapper = mount(<DownloadButton
            title="Download"
            onClick={onClickMock}
            disabled={true}
            fontSize="small"
        ></DownloadButton>);

        wrapper.find(IconButton).props().onClick()
        expect(onClickMock).toHaveBeenCalledTimes(1);
    });

    it('the button should be disabled', () => {
        const wrapper = mount(<DownloadButton
            title="Download"
            onClick={() => { }}
            disabled={true}
            fontSize="small"
        ></DownloadButton>);

        expect(wrapper.find(IconButton).props().disabled).toBe(true);
    });

    it('the button should be enabled', () => {
        const wrapper = mount(<DownloadButton
            title="Download"
            onClick={() => { }}
            disabled={false}
            fontSize="small"
        ></DownloadButton>);

        expect(wrapper.find(IconButton).props().disabled).toBe(false);
    });

    it('the color, fontSize and size props should be equal to the default values if they are not provided', () => {
        const wrapper = mount(<DownloadButton
            title="Download"
            onClick={() => { }}
            disabled={false}
            fontSize="small"
        ></DownloadButton>);

        expect(wrapper.props().color).toBe('theme');
        expect(wrapper.props().fontSize).toBe('small');
        expect(wrapper.props().size).toBe('medium');
    });
});