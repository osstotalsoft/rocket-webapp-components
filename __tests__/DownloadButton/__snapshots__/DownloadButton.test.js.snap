// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadButton snapshot 1`] = `
<span
  aria-describedby={null}
  className=""
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title="Download"
>
  <button
    aria-label="Download"
    className="MuiButtonBase-root MuiIconButton-root makeStyles-button-1 makeStyles-theme-17 Mui-disabled Mui-disabled"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM17 13l-5 5-5-5h3V9h4v4h3z"
        />
      </svg>
    </span>
  </button>
</span>
`;
